### ISSUE TITLE
Multilevel with ordinal variables

### ISSUE THREAD

**Issue by RMKirkpatrick** [OpenMx/OpenMx#241]
See https://openmx.ssri.psu.edu/node/4536 .  I thought trying to use the multilevel feature with threshold variables would cause an error, but apparently someone got it to actually run, and return what is probably nonsense.  Bug?

Obviously, multilevel modeling with threshold variables would be a nice thing to have...

**Comment by RMKirkpatrick** [comment]
Multilevel regression modeling with ordinal variables (or any other kind of variable with which the identity link isn't used) could be accomplished either by integrating out the random effects by way of quadrature, or by MCMC Bayes (probably through an interface to Stan).  The former option is less general and would be really slow with high-dimensional random effects, but would be less work to implement, and would not be so terribly different from what OpenMx already does.  The latter option would be quite general and could still feasibly work in high dimensions, but would be more difficult to implement, and would be a step in a very different direction for OpenMx.  Figuring out the UI for Bayes would be particularly challenging.

**Comment by mcneale** [comment]
Agreed.  Would pretending that the data are continuous and computing MLR (robust) SEs be a slightly-better-than-nothing stop-gap?

**Comment by RMKirkpatrick** [comment]
> Agreed. Would pretending that the data are continuous and computing MLR (robust) SEs be a slightly-better-than-nothing stop-gap?

Perhaps.  But `imxRobustSE()` does not handle multilevel data correctly.  Its man page says as much.

**Comment by mcneale** [comment]
Yeah, I figured as much... how effortful to revise, compared to your option 1?  Option 2 needs its own grant.   Or someone in big oil to start donating...

**Comment by RMKirkpatrick** [comment]
> Yeah, I figured as much... how effortful to revise, compared to your option 1?

Trying to revise `imxRobustSE()` to work properly with multilevel data would be a waste of time.  We would be better off implementing robust SEs in the backend, where the dataset has been flattened.

**Comment by jpritikin** [comment]
[Pairwise maximum likelihood](https://bpspsychub.onlinelibrary.wiley.com/doi/full/10.1111/bmsp.12243) is an approach that works with small cluster sizes. It could also be useful for non-multilevel models when the number of ordinal variables is between 12-30. PML is like a compromise between WLS and ML.
